File: apps/web/src/sentry.ts
import * as Sentry from "@sentry/react";

export function initSentry(): boolean {
  const dsn = import.meta.env.VITE_SENTRY_DSN;
  if (!dsn) return false;
  Sentry.init({
    dsn,
    environment: import.meta.env.MODE,
    tracesSampleRate: 0.1,
  });
  return true;
}

File: apps/api/src/api/sentry.py
import os

import sentry_sdk


def init_sentry() -> bool:
    dsn = os.environ.get("SENTRY_DSN")
    if not dsn:
        return False
    sentry_sdk.init(
        dsn=dsn,
        traces_sample_rate=0.1,
        environment=os.environ.get("ENV", "local"),
        release=os.environ.get("FLY_RELEASE_VERSION", "dev"),
    )
    return True
